src/aggregator/aggregator/map.go

```diff
@@ -340,8 +346,27 @@ func (m *metricMap) tick(target time.Duration) tickResult {
 		numTimedActive       int
 		numTimedExpired      int
 		entryIdx             int
+
+		done bool
 	)
+
+	// NB: if no doneChan provided, do not interrupt the tick.
+	if doneCh == nil {
```

Comment:
when is a done chan not provided?

Reply:
It shouldn't be if everything is hooked up correctly, this is largely just to be defensive here; might remove it and hard panic or better yet error if we don't have a chan passed in to make it more explicit

Reply:
Huh, actually... TIL that you can use nil channels in a select https://play.golang.org/p/DltRooS3v7D